packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-inline-checkout-button-strategy.ts

```diff
@@ -0,0 +1,334 @@
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+export default class PaypalCommerceInlineCheckoutButtonStrategy implements CheckoutButtonStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _paypalCommerceRequestSender: PaypalCommerceRequestSender,
+        private _orderActionCreator: OrderActionCreator,
+        private _consignmentActionCreator: ConsignmentActionCreator,
+        private _billingAddressActionCreator: BillingAddressActionCreator,
+        private _paymentActionCreator: PaymentActionCreator
+    ) {}
+
+    async initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const { methodId, paypalcommerceinline } = options;
+        const {
+            acceleratedCheckoutContainerDataId,
+            buttonContainerDataId,
+            buttonContainerClassName,
+            checkoutNowElementDataId,
+            style,
+            onComplete,
+        } = paypalcommerceinline || {};
+
+        if (!methodId) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.methodId" argument is not provided.');
+        }
+
+        if (!acceleratedCheckoutContainerDataId) {
+            throw new InvalidArgumentError(`Unable to initialize payment because "options.paypalcommerceinline.acceleratedCheckoutContainerDataId" argument is not provided.`);
+        }
+
+        if (!buttonContainerDataId) {
+            throw new InvalidArgumentError(`Unable to initialize payment because "options.paypalcommerceinline.buttonContainerDataId" argument is not provided.`);
+        }
+
+        if (!checkoutNowElementDataId) {
+            throw new InvalidArgumentError(`Unable to initialize payment because "options.paypalcommerceinline.checkoutNowElementDataId" argument is not provided.`);
+        }
+
+        if (!onComplete || typeof onComplete !== 'function') {
+            throw new InvalidArgumentError(`Unable to initialize payment because "options.paypalcommerceinline.onComplete" argument is not provided or it is not a function.`);
+        }
+
+        await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
+        await this._store.dispatch(this._consignmentActionCreator.loadShippingOptions());
+
+        const state = this._store.getState();
+        const currency = state.cart.getCartOrThrow().currency.code;
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const paypalCommerceSdk = await this._paypalScriptLoader.getPayPalSDK(paymentMethod, currency, false);
+
+        if (!paypalCommerceSdk) {
+            throw new PaymentMethodClientUnavailableError();
+        }
+
+        this._renderButton(
```

Comment:
Can we use an object with interface :D

Reply:
You are right... It might be easier to read then